### PR TITLE
terraform-providers: switch to hash/vendorHash

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
@ofborg build terraform.full

**Comment by zowoq** [comment]
No new failures.

File: pkgs/applications/networking/cluster/terraform-providers/update-provider [hunk 1]
```diff
@@ -93,14 +93,14 @@ repo_root=$(git rev-parse --show-toplevel)
 
 generate_hash() {
   nix-prefetch -I nixpkgs="${repo_root}" \
-    "{ sha256 }: (import ${repo_root} {}).terraform-providers.${provider}.$1.overrideAttrs (_: { $2 = sha256; })"
+    "{ sha256 }: (import ${repo_root} {}).terraform-providers.${provider}.$1.overrideAttrs (_: { inherit sha256; })"
```
[the remark this answers is not in the document]
Comment (zowoq) [reply pkgs/applications/networking/cluster/terraform-providers/update-provider:96]: This doesn't seem to work with anything but `sha256`?

File: pkgs/applications/networking/cluster/terraform-providers/update-provider [hunk 2]
```diff
@@ -93,14 +93,14 @@ repo_root=$(git rev-parse --show-toplevel)
 
 generate_hash() {
```
[the remark this answers is not in the document]
Comment (zowoq) [reply pkgs/applications/networking/cluster/terraform-providers/update-provider:94]: Don't think we can, AFAIK the hash type is whatever `nix` defaults to. This script was pinned to the same branch for its NIX_PATH but had to drop that to get the `updateScript` to work.

**Review by zimbatm — APPROVED** [review]

**Pull request merged**